### PR TITLE
Added Device class representing Data-API notion of device

### PR DESCRIPTION
**Pull request by oleksandr-pavlyk**
Data-API notion of device is higher level abstraction. The class
does allow for public constructor, has class method to create
instance instead.

It typesets as Device(filter_string) for parent devices and
Device(queue) for sub-devices. It can constructed from

   - filter selector string
   - SyclDevice corresponding to a root device
     (attempt at passing sub-device raises and error)
   - SyclQueue
   - Another instance of Device class

It implements sycl_queue, sycl_device, sycl_context attributes.

usm_ndarray adds .device property, and .to_device(device) method.

```
In [5]: X = dpt.usm_ndarray((4, 5), dtype='d')

In [6]: X.device
Out[6]: Device(level_zero:gpu:0)

In [7]: Y = X.to_device('cpu')

In [8]: Y.device
Out[8]: Device(opencl:cpu:0)
```

### CODE REVIEW

File: dpctl/tensor/_device.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import dpctl
```
Comment (diptorupd) [inline dpctl/tensor/_device.py:1]: license header is needed.

File: dpctl/tensor/_device.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+
+class Device:
+    def __new__(clf, *args, **kwargs):
```
Comment (diptorupd) [inline dpctl/tensor/_device.py:5]: Please document the class and the functions.

Comment (diptorupd) [inline dpctl/tensor/_device.py:5]: `s/clf/cls/`

File: dpctl/tensor/_device.py (hunk 1 above)
Comment (oleksandr-pavlyk) [reply dpctl/tensor/_device.py:16]: Added.

**Review by diptorupd — APPROVED** [review]

**Pull request merged**